### PR TITLE
Add a "pt2 compliant" tag; add config to graph break on non-pt2_compliant ops

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #112053
* __->__ #111933
* #111948
* #111915
* #111912

This PR:
- adds the pt2 compliant tag. This tag specifies that the operator works
  with the PT2 compilation APIs. A custom op author should test their
  ops with opcheck if they choose to add this tag.
- adds a config for Dynamo to allow only pt2 compliant ops into the
  graph and graph break on all other OpOverload/OpOverloadPacket.

Bikeshedding help wanted on the name of the tag. It should be easily
grep-able so we can set up rules for it.

Test Plan:
- new tests

Reviewers:

Subscribers:

Tasks:

Tags:

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: aten/src/ATen/native/tags.yaml [hunk 1]
```diff
@@ -3,6 +3,15 @@
 - tag: inplace_view
   desc: |
           This tag indicates if an operator *only* modifies the tensor metadata
+- tag: pt2_compliant_tag
```
Comment (zou3519) [inline aten/src/ATen/native/tags.yaml:6]: torch.Tag.pt2_compliant_tag and at::Tag::pt2_compliant_tag may sound a bit goofy, but they are easily greppable

**Comment by ezyang** [comment]
Bikeshed is fine.

Are you going to grandfather aten ops into this, or treat the aten namespace specially?

**Comment by zou3519** [comment]
> Bikeshed is fine.
> 
> Are you going to grandfather aten ops into this, or treat the aten namespace specially?

Plan is to grandfather all aten ops created via codegen into this by automatically adding the tag during codegen

**Comment by ezyang** [comment]
no, it's not necessary

**Review by ezyang — APPROVED** [review]

File: torch/_dynamo/variables/torch.py [hunk 2]
```diff
@@ -148,6 +148,37 @@ def _dynamo_overriden_transformers_eq(self, other):
...
+err_epilogue = (
+    "With the current config, we will graph break "
+    "(and fall back to eager-mode PyTorch) on all ops "
+    "that have do not have the 'pt2_compliant_tag'. "
+    "Please see the following doc for how to mark this op as PT2 compliant "
+    "https://docs.google.com/document/d/1W--T6wz8IY8fOI0Vm8BF44PdBgs283QvpelJZWieQWQ"
+)
+
+
+def check_allowed_op(value):
```
Comment (yanboliang) [inline torch/_dynamo/variables/torch.py:160]: I think we have more elegant way to handle this after #111622.

**Pull request closed without merging**